src/embezzler.ts

```diff
@@ -49,46 +51,131 @@ import { Macro, shouldRedigitize, withMacro } from "./combat";
 import { usingThumbRing } from "./dropsgear";
 import { crateStrategy, equipOrbIfDesired } from "./extrovermectin";
 import { baseMeat, globalOptions, ltbRun, WISH_VALUE } from "./lib";
-import { determineDraggableZoneAndEnsureAccess, draggableFight } from "./wanderer";
+import { familiarWaterBreathingEquipment, waterBreathingEquipment } from "./outfit";
+import { determineDraggableZoneAndEnsureAccess, DraggableFight } from "./wanderer";
 
-type EmbezzlerFightOptions = {
-  location?: Location;
-  macro?: Macro;
-};
+/**
+ * Configure the behavior of the fights in use in different parts of the fight engine
+ * @interface EmbezzlerFightConfigOptions
+ * @member {Requirement[]?} requirements maximizer requirements to use for this fight (defaults to empty)
+ * @member {draggableFight?} draggable if this fight can be pulled into another zone and what kind of draggable it is (defaults to undefined)
+ * @member {boolean?} canInitializeWandererCounters if this fight can be used to initialize wanderers (defaults to true)
+ * @member {boolean?} gregariousReplace if this is a "monster replacement" fight - pulls another monster from the CSV (defautls to false)
+ * @member {boolean?} wrongEncounterName if mafia does not update the lastEncounter properly when doing this fight (defaults to value of gregariousReplace)
+ */
+interface EmbezzlerFightConfigOptions {
+  requirements?: Requirement[];
+  draggable?: DraggableFight;
+  canInitializeWandererCounters?: boolean;
+  wrongEncounterName?: boolean;
+  gregariousReplace?: boolean;
+}
+
+class EmbezzlerFightRunOptions {
```

Comment:
I'm assuming we have a wrapper class here per our discussion about frequently using lines like `const macro = options.macro ?? embezzlerFightMacro()`, but don't see how this resolves the issue. It feels like overkill to make this little class, although a docstring would go a long way to making this feel less unnecessary

Reply:
Basically:

What i wanted people to be able to do is `options.location` within the EmbezzlerFight without having to check for null or undefined. In order to do that, location needs to be not null/undefined. In order for it to be not null or undefined, it needs to have value. I was not comfortable giving a nonsense default value for non wandering EmbezzlerFights - I would prefer those error out if they try to access the location. The easiest way to accomplish both of these things was wrapping it in a class with a getter that would raise if location was undefined.